### PR TITLE
Migrate core to TS p4

### PR DESCRIPTION
**Pull request by albertogasparin**
This is the last PR with lots of renaming for `loot-core`!

### CODE REVIEW

File: packages/loot-core/src/server/api.ts
Comment (albertogasparin) [inline packages/loot-core/src/server/api.ts:196]: Minor runtime change, as `getDownloadError` does not expect 2 arguments

File: packages/loot-core/src/client/SpreadsheetProvider.tsx
Comment (albertogasparin) [inline packages/loot-core/src/client/SpreadsheetProvider.tsx:80]: Minor runtime change, as `get` does not expect 3 arguments

File: packages/loot-core/src/server/db/index.ts
Comment (albertogasparin) [inline packages/loot-core/src/server/db/index.ts:638]: Minor runtime change so we can have TS complain if we pass an extra argument

File: packages/loot-core/src/server/main.ts
Comment (albertogasparin) [inline packages/loot-core/src/server/main.ts:1871]: Minor runtime change because `replace` does nothing

File: packages/loot-core/src/server/main.ts
Comment (albertogasparin) [inline packages/loot-core/src/server/main.ts:1923]: Minor runtime change because `loadBudget` `showUpdate` does nothing

File: packages/loot-core/src/types/models/payee.d.ts
Comment (albertogasparin) [inline packages/loot-core/src/types/models/payee.d.ts:9]: @j-f1 @MatissJanis  I'd recommend either of you setting some time to properly think about these model types. I tried to get my head around them but I'm still confused 😅 . There are DB fields, models with snake_case, models with camelCase, hydrated models, client side models, migration models, ... I got lost.

Right now I've added this stopgap to all `*Entity` stuff so I could unblock myself, but I'd like you to use your knowledge of the system to come up (in another PR) with 2-3 proper, complete interfaces for each model that we can use everywhere.

File: packages/loot-core/src/client/platform.ts
Comment (j-f1) [inline packages/loot-core/src/client/platform.ts:12]: Future task: since we deleted the mobile codebase this will always be `web`

File: packages/loot-core/src/platform/server/fs/index.d.ts
Comment (j-f1) [inline packages/loot-core/src/platform/server/fs/index.d.ts:52]: 💯

File: packages/loot-core/src/platform/server/fs/index.electron.ts
Comment (j-f1) [inline packages/loot-core/src/platform/server/fs/index.electron.ts:129]: Can you add a comment above this one about why the any is necessary?

File: packages/loot-core/src/server/encryption-internals.ts
Comment (j-f1) [inline packages/loot-core/src/server/encryption-internals.ts:77]: Why are the args in different orders?

File: packages/loot-core/src/server/cloud-storage.ts
Comment (MatissJanis) [inline packages/loot-core/src/server/cloud-storage.ts:282]: Just double-checking: will this actually work?

File: packages/loot-core/src/server/main.ts
Comment (MatissJanis) [inline packages/loot-core/src/server/main.ts:151]: Is `'transactions-filter'` even used? Couldn't find where, so perhaps worth removing entirely.

File: packages/loot-core/src/server/cloud-storage.ts
Comment (albertogasparin) [reply packages/loot-core/src/server/cloud-storage.ts:282]: That function is a misnomer as it returns an object, not an error. Also it is used like that (without `new`) in few other places, so I guess it is fine.

**Review by MatissJanis — APPROVED** [review]

**Review by j-f1 — APPROVED** [review]

**Comment by albertogasparin** [comment]
Thanks for approving! I thought I could merge this now but I still get "Only those with [write access](https://docs.github.com/articles/what-are-the-different-access-permissions) to this repository can merge pull requests." 🤔 So I guess one of you needs to press the button

**Review by j-f1 — APPROVED** [review]
Seems like it didn’t pick up the approvals… maybe a bug on their end??

**Comment by MatissJanis** [comment]
Added more permissions for you @albertogasparin . Try now.

**Pull request merged**
